### PR TITLE
Add anonymize user integration test

### PR DESCRIPTION
**Pull request by gtrkiller**
This PR adds an integration test contemplating the case when an email in the middle of the list for the anonymize action fails to be anonymized.

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -134,6 +136,13 @@ async def add_admin(app: Application):
     assert action.status == "completed"
     assert action.results["user"] == email
     assert f'Admin with email "{email}" correctly created' in action.results["output"]
+    action2: juju.action.Action = await app.units[0].run_action(  # type: ignore
```
[the remark this answers is not in the document]
Comment (gtrkiller) [reply tests/integration/test_charm.py:139]: I think that, as we are executing the same thing as in the first action, the first comment applies for both type: ignore clauses. Would it be advisable to add a comment again anyway?

**Review by amandahla — APPROVED** [review]
LGTM

**Review by yanksyoon — APPROVED** [review]
LGTM

File: tests/integration/test_charm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdkandersson) [reply tests/integration/test_charm.py:139]: potentially since someone might just be reading this particular part of the code

**Pull request merged**